ui/docs/components.md

```diff
@@ -0,0 +1,61 @@
+# Writing and consuming components
+
+Components can range from small, highly reusable "atoms" to large units with lots of business logic specific to one workflow or action. In any scenario, these are things to keep in mind while developing components for the Vault UI.
+
+Please note that these guidelines are aspirational and you will see instances of antipatterns in the codebase. Many of these should be updated as we move forward. As with any ruleset, sometimes it is appropriate to break the rule.
+
+## Page components for every route
+
+Route templates should render a `Page` component, which includes breadcrumbs, page title, and then renders whatever else should be on the page (often another scoped component).
+
+- This component should be named something like `<Page::CreateFoo />` and you can create it like `ember g component page/create-foo -gc`.
+- The Route should pass the model hook data into the component in the template. However, if the model hook returns multiple objects they should each be passed into the Page component as separate args. For example: within a route's template whose model hook returns two different data models, the route's template would look like:
+
+```hbs
+<Page::CreateFoo @config={{this.model.config}} @foo={{this.model.foo}} />
+```
+
```

Comment:
When I first went into this codebase, I was trying to figure out when to create `Page::Components` vs regular components in the engine's `addon/components`. I think this section provides a lot of clarity as to when to make a Page component vs a regular scoped component!